RELEASES.md

```diff
@@ -0,0 +1,414 @@
+# Version 2.0 Syntax (2019-11-11)
+
+The ink! version 2.0 syntax has one major philosophy:
+
+> Just. Be. Rust.
+
+To accomplish this, we take advantage of all the standard Rust types and structures and use
+[attribute macros](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) to
+tag these standard structures to be different parts of the ink! language.
+
+Anything that is not tagged with an `#[ink(...)]` attribute tag is just standard Rust, and can be
+used in and out of your contract just like standard Rust could be used!
+
+**Every valid ink! contract is required to have at least one `#[ink(constructor)]`, at least one
+`#[ink(message)]` and exactly one `#[ink(storage)]` attribute.**
+
+Follow the instructions below to understand how to migrate your ink! 1.0 contracts to this new ink!
+2.0 syntax.
+
+## Declaring a Contract
+
+The fundamental change with the new ink! syntax is how we declare a new contract.
+
+We used to wrap the whole ink! contract into a `contract!` macro. At that point, all syntax within
+the macro could be custom, and in our first iteration of the language, we used that in ways that
+made our code not really Rust anymore.
+
+Now we wrap the whole contract in a standard Rust module, and include an attribute tag to identify
+this object as part of the ink! language. This means that all of our code from this point forward
+will be valid Rust!
+
+<table style="width: 100%;">
+<tr>
+<th>Before</th>
+<th>After</th>
+</tr>
+<tr>
+<td>
+
+```rust
+contract! {
+    ...
+}
+```
+
+</td>
+<td>
+
+```rust
+#[ink::contract(version = "0.1.0")]
```

Comment:
Technically it is `ink_lang2::contract` but since we import `ink_lang2` using `use ink_lang2 as ink;` above this works.

Reply:
Will it always be `ink_lang2`? Or will you rename it once you refactor syntax 1 out?

Reply:
We might want to rename it, yes. I was even thinking about introducing a shim-crate `ink` that covers all other crates so that contract authors no longer have to include X different crates but just this single one instead.

Reply:
Names are seldomly final in ink! :P

Reply:
Well... dunno if you want me to act on the change or not :P

Reply:
Okay sorry, I think we should include the `use` statement to make it clear what happened on this line. E.g.:
```
use ink_lang2 as ink;

#[ink::contract(version = "0.1.0")]
mod foo {
    ...
}
```

Reply:
```suggestion
use ink_lang2 as ink;

#[ink::contract(version = "0.1.0")]
```

Reply:
Done :)